### PR TITLE
Linux chapter beyond the basics

### PR DESCRIPTION
**Pull request by redfast00**

### CODE REVIEW

File: intro-Linux/ch_beyond_the_basics.tex [hunk 1]
```diff
@@ -5,20 +5,20 @@ \chapter{Beyond the basics}
...
- \item ``stdout'': standard output channel, for regular output
- \item ``stderr'': standard error channel, for errors \& warnings
+ \item \verb|stdout|: standard output channel, for regular output
+ \item \verb|stderr|: standard error channel, for errors and warnings
 \end{enumerate}
 
 \subsection{Redirecting ``stdout''}
 
-``>'' writes the (stdout) output of a command to a file and ''overwrites''
+\verb|>| writes the (\verb|stdout|) output of a command to a file and ''overwrites''
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: use `\emph` rather than `''...''`?

File: intro-Linux/ch_beyond_the_basics.tex [hunk 2]
```diff
@@ -227,63 +232,66 @@ \section{Counting: ``wc''}
 
 \section{Searching file contents: ``grep''}
 
-``grep'' is such an important command. It was originally an abbreviation for
+\verb|grep| is such an important command. It was originally an abbreviation for
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: drop the `such` here? sounds a bit weird to me...

File: intro-Linux/ch_beyond_the_basics.tex [hunk 3]
```diff
@@ -227,63 +232,66 @@ \section{Counting: ``wc''}
...
+example, to pull the first column (\verb|-f 1|, the first \strong{f}ield)
+from (an unquoted) csv (comma-separated values, so \verb|-d ','|: \strong{d}elimited by \verb|,|)
+file, you can use the following:
 
 \begin{prompt}
 %\shellcmd{cut -f 1 -d ',' mydata.csv}%
 \end{prompt}
 
-\section{``sed''} ``sed'' is the stream editor. It is used to replace text in a
+\section{``sed''}
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `\texttt`?

File: intro-Linux/ch_beyond_the_basics.tex [hunk 4]
```diff
@@ -301,16 +309,15 @@ \section{Basic Shell Scripting}
...
 \subsection{Shebang}
-The first line of the script is the so called shebang (``\#'' is sometimes called
-hash and ``!'' is sometimes called bang). This line tells the shell which
+The first line of the script is the so called shebang (\verb|#| is sometimes called
+hash and \verb|!| is sometimes called bang). This line tells the shell which
 command should execute the script. In the most cases this will simply be the
 shell itself. The line itself looks a bit weird, but you can copy paste this
 line as you need not worry about it further. It is however very important this
-is the very first line of the script!
+is the very first line of the script! These are all valid shebangs:
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: Maybe mention that only *one* of these should be used?

Also make them separate boxes to clarify that?

File: intro-Linux/ch_beyond_the_basics.tex [hunk 5]
```diff
@@ -332,13 +339,18 @@ \subsection{Conditionals}
...
+    echo "it exists"
+ fi
 \end{code}
 
-so the basic structure is
+Or only if a certain variable is bigger than one
 
 \begin{code}{bash}
- if [ $AANTAL -eq 1 ]
+ if [ $AMOUNT -gt 1 ]
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: Ugh, nice catch... Talk about confusing readers :)

File: intro-Linux/ch_beyond_the_basics.tex [hunk 6]
```diff
@@ -349,11 +361,10 @@ \subsection{Conditionals}
 brackets, the \strong{then} needs to be on the beginning of a line. It is best
 to just copy this example and modify it.
 
-In the initial example I used -d to test if a directory existed. There are
-several more checks which can be found here:
-\url{http://tldp.org/LDP/Bash-Beginners-Guide/html/sect_07_01.html}
+In the initial example I used \verb|-d| to test if a directory existed. There are
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `I` -> `we`

File: intro-Linux/ch_beyond_the_basics.tex [hunk 7]
```diff
@@ -381,41 +392,44 @@ \subsection{Loops}
...
+command like this:
 
 \begin{prompt}
-command 2$>$&1 output.log   # one single output file, both output and errors
+command 2>&1 output.log   # one single output file, both output and errors
 \end{prompt}
 
-so you add "2>\&1 output.log" at the end of any command and it will combine
-output and error output into a single file for you to inspect named output.log.
+If you add \verb|2>&1 output.log| at the end of any command and it will combine
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: one `and` too many here?

File: intro-Linux/ch_beyond_the_basics.tex [hunk 8]
```diff
@@ -381,41 +392,44 @@ \subsection{Loops}
...
 
 \begin{prompt}
-command $>$ output.log 2$>$ output.err  # errors in a separate file
+command > output.log 2> output.err  # errors in a separate file
 \end{prompt}
 
-this will write regular output to output.log and error output to output.err.
+this will write regular output to \verb|output.log| and error output to \verb|output.err|.
+
+You can then look for the errors with `less` or search for specific text with `grep`.
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `\verb` for `less` & `grep`?

File: intro-Linux/ch_beyond_the_basics.tex [hunk 9]
```diff
@@ -485,55 +498,56 @@ \subsection{PBS pragmas}
 how much memory will it use, how long will it run? These things can be specified
 inside a script with what we call PBS pragmas.
 
+This option tells PBS to use 1 node and 1 core:
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `option` -> `pragma` (or `speciall comment`)?

File: intro-Linux/ch_beyond_the_basics.tex [hunk 10]
```diff
@@ -485,55 +498,56 @@ \subsection{PBS pragmas}
...
+
 \begin{code}{bash}
   #PBS -l nodes=1:ppn=1   # single-core
 \end{code}
 
 For parallel software, you can request multiple cores (OpenMP) and/or multiple
-nodes (MPI). '''Only use this when the software you use is capable of working in
-parallel.'''
+nodes (MPI). \strong{Only use this when the software you use is capable of working in
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `strong` -> `emph`?

File: intro-Linux/ch_beyond_the_basics.tex [hunk 11]
```diff
@@ -485,55 +498,56 @@ \subsection{PBS pragmas}
...
+parallel.}. Here is an example:
 
 \begin{code}{bash}
   #PBS -l nodes=1:ppn=16  # single-node, multi-core
   #PBS -l nodes=5:ppn=16  # multi-node
 \end{code}
 
-This option tells PBS to use 1 node and 1 core.
+We submit it on the long queue:
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `We intend to submit`

File: intro-Linux/ch_beyond_the_basics.tex [hunk 12]
```diff
@@ -485,55 +498,56 @@ \subsection{PBS pragmas}
...
 
 \begin{code}{bash}
   #PBS -q long
 \end{code}
 
-We submit it on the long queue.
+We request a total running time of 2 days (48 hours).
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `48 hours (2 days)`

File: intro-Linux/ch_beyond_the_basics.tex [hunk 13]
```diff
@@ -485,55 +498,56 @@ \subsection{PBS pragmas}
...
 
 \begin{code}{bash}
   #PBS -l walltime=48:00:00
 \end{code}
 
-We request a total running time of 2 days (48 hours).
+We tell PBS the name of our job:
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: `We specify the desired name of our job:`

File: intro-Linux/ch_beyond_the_basics.tex [hunk 14]
```diff
@@ -485,55 +498,56 @@ \subsection{PBS pragmas}
...
 \begin{code}{bash}
   #PBS -m abe
 \end{code}
 
-This specifies mail options.
 \begin{enumerate}
-\item \strong{a} means mail is sent when the job is aborted by the batch system.
-\item \strong{b} means mail is sent when the job begins.
-\item \strong{e} means mail is sent when the job ends.
+\item \verb|a| means mail is sent when the job is aborted by the batch system.
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: remove the `by the batch system` (readers may start wondering what a `batch system` is :))

File: intro-Linux/ch_beyond_the_basics.tex (hunk 8 above)
Comment (redfast00) [reply intro-Linux/ch_beyond_the_basics.tex]: Whoops, confused markdown and latex

File: intro-Linux/ch_beyond_the_basics.tex [hunk 15]
```diff
@@ -485,55 +503,56 @@ \subsection{PBS pragmas}
...
 
-We submit it on the long queue.
+We request a total running time of 48 hours (2 days).
 
 \begin{code}{bash}
   #PBS -l walltime=48:00:00
 \end{code}
 
-We request a total running time of 2 days (48 hours).
+We tell specify a desired name of our job:
```
Comment (boegel) [inline intro-Linux/ch_beyond_the_basics.tex]: drop the `tell`

Comment (redfast00) [reply intro-Linux/ch_beyond_the_basics.tex]: I probably need to take a break...

**Pull request merged**